api/command_join.go

```diff
@@ -42,18 +42,11 @@ func (me *JoinProvider) DoCommand(c *Context, channelId string, message string)
 
 			if v.Name == message {
 
-				if v.Type == model.CHANNEL_DIRECT {
+				if v.Type != model.CHANNEL_OPEN {
 					return &model.CommandResponse{Text: c.T("api.command_join.fail.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+				} else {
```

Comment:
No need for the `else` here since it's the end of the function as you're returning. Can you remove it?